install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -686,12 +670,36 @@ data:
 {{- if and .Values.endpointRoutes .Values.endpointRoutes.enabled }}
   enable-endpoint-routes: {{ .Values.endpointRoutes.enabled | quote }}
 {{- end }}
-{{- if .Values.cni.configMap }}
+
+  ## CNI configuration
+{{- if and .Values.cni.install  (not .Values.cni.customConf)  }}
```

Comment:
Uh oh, you're right. Oops!

Reply:
Fortunately, nothing sets `cni.readCniConf` in the Cilium documentation. In any case, the previous behavior is preserved.

Reply:
I think one usecase for `readCniConf` is to set per-node IPAM parameters, I think we might have a few users setting it. But since the old behavior is preserved, that will remain working. 

Long term, your per-node configuration mechanism (unrelated to this PR) might be a better alternative to setting per-node IPAM parameters in the CNI config as well.